src/category_theory/simplex_category.lean

```diff
@@ -0,0 +1,252 @@
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  simp with push_cast,
+  split_ifs; { simp at *; linarith, },
+end
+
+/-- The second part of the third simplicial identity -/
+lemma δ_comp_σ_succ {n} {i : fin (n+1)} :
+  δ i.succ ≫ σ i = 𝟙 _ :=
+begin
+  ext j,
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  simp with push_cast,
+  split_ifs; { simp at *; linarith, },
+end
+
+/-- The fourth simplicial identity -/
+lemma δ_comp_σ_of_gt {n} {i : fin (n+2)} {j : fin (n+1)} (H : j.cast_succ < i) :
+  δ i.succ ≫ σ j.cast_succ = σ j ≫ δ i :=
+begin
+  ext k,
+  dsimp [δ, σ, fin.succ_above, fin.pred_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  rcases k with ⟨k, _⟩,
+  simp at H,
+  -- rw apply_dite fin.cast_succ,
+  simp [apply_dite fin.cast_succ] with push_cast, -- `simp?` doesn't work here
+  split_ifs,
+  -- Hope for the best from `linarith`:
+  any_goals { simp at *, try { linarith }, },
+  -- Four of the goals need special handling:
+  { simp at h_1,
+    exact lt_irrefl (k - 1) (lt_of_lt_of_le
+     (nat.pred_lt (ne_of_lt (lt_of_le_of_lt (zero_le _) h_1)).symm)
+     (le_trans (nat.le_of_lt_succ h) h_2)) },
+  { simp at h_1, linarith, },
+  { exfalso, exact lt_irrefl _ (lt_of_le_of_lt (nat.le_pred_of_lt (nat.lt_of_succ_le h)) h_3), },
+  { exact (nat.succ_pred_eq_of_pos (lt_of_le_of_lt (zero_le _) h_2)).symm, },
+end
+
+local attribute [simp] fin.pred_mk
+
+/-- The fifth simplicial identity -/
+lemma σ_comp_σ {n} {i j : fin (n+1)} (H : i ≤ j) :
+  σ i.cast_succ ≫ σ j = σ j.succ ≫ σ i :=
+begin
+  ext k,
+  dsimp [σ, fin.pred_above],
+  rcases i with ⟨i, _⟩,
+  rcases j with ⟨j, _⟩,
+  rcases k with ⟨k, _⟩,
+  simp at H,
+  simp with push_cast,
+  split_ifs,
+  -- Hope for the best from `linarith`:
+  any_goals { simp at *, try { linarith }, },
+  { exact false.elim
+    (lt_irrefl (k - 1)
+      (lt_of_lt_of_le (nat.pred_lt (id (ne_of_lt (lt_of_le_of_lt (zero_le i) h)).symm))
+        (le_trans h_2 (nat.succ_le_of_lt h_1)))) },
+  { exact false.elim
+    (lt_irrefl j (lt_of_lt_of_le (nat.pred_lt_pred (nat.succ_ne_zero j) h_2) h_1)) },
+end
+
+end generators
+
+section skeleton
+
+lemma skeletal : skeletal simplex_category :=
+λ X Y I,
+begin
+  rcases I with ⟨I⟩,
+  suffices : fintype.card (fin (X+1)) = fintype.card (fin (Y+1)),
+  { simpa, },
+  { apply fintype.card_congr,
+    refine ⟨I.hom, I.inv, _, _⟩,
+    intro x, exact congr_fun (congr_arg (preorder_hom.to_fun) I.hom_inv_id : _) x,
+    intro y, exact congr_fun (congr_arg (preorder_hom.to_fun) I.inv_hom_id : _) y, }
+end
+
+/-- The functor that exhibits `simplex_category` as skeleton
+of `NonemptyFinLinOrd` -/
+def skeletal_functor :
+  simplex_category ⥤ NonemptyFinLinOrd.{u} :=
+{ obj := λ n, NonemptyFinLinOrd.of $ ulift (fin (n+1)),
+  map := λ m n f, ⟨λ i, ⟨f i.down⟩, λ ⟨i⟩ ⟨j⟩ h, show f i ≤ f j, from f.monotone h⟩, }
+
+namespace skeletal_functor
+
+instance full : full.{0 u 0 (u+1)} skeletal_functor.{u} :=
```

Comment:
The two instances below don't have/need a name spelt out. Does this one need it, for some reason?

Reply:
Nope. Also removed the unnecessary universe annotations.